mobile/ios/WirelessDebug/WirelessDebugger/WebSocketMessenger.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  WebSocketMessenger.swift
+//  Wireless Debug iOS Test App
+//
+
+import UIKit
+import Foundation
+
+/// Handles opening a WebSocket connection and sending the necessary messages
+/// across that connection.
+class WebSocketMessenger {
+    var webSocket = WebSocket()
+    var logsToSend: [String] = []
+    var connectionRetries = 0
+
+    /// Create a WebSocket connection to the given socket address and API Key.
+    ///
+    /// - Parameters:
+    ///   - socketAddress: the WebSocket address to connect to
+    ///   - apiKey: the API Key provided by the web interface
+    init(_ socketAddress: String, apiKey: String) {
+        // When the WebSocket connection opens, send a startSession message.
+        self.webSocket.event.open = {
+            NSLog("WebSocket Connection to \(socketAddress) Opened")
+
+            self.send("startSession", data: [
+                "osType": "iOS",
+                "apiKey": apiKey,
+                "deviceName": UIDevice.current.name,
+                "appName": Bundle.main.infoDictionary!["CFBundleName"] as! String,
+            ])
+            self.connectionRetries = 0
+        }
+
+        // If the connection closes, try to reconnect 10 times.
+        self.webSocket.event.close = { code, reason, clean in
```

Comment:
When it's sending the logs, it can:
1) Reopen the socket if its not already open
2) Send up the logs

There's no need for retrying at this layer. The caller, LogStreamer can redispatch sendLogs if there is an error.